src/AaveV2CollectorContractConsolidation.sol

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import {TokenAddresses} from "./TokenAddresses.sol";
+import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {AggregatorV3Interface} from "./external/AggregatorV3Interface.sol";
+import {ERC20} from "@openzeppelin/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/token/ERC20/utils/SafeERC20.sol";
+
+/// @title AaveV2CollectorContractConsolidation
+/// @author Llama
+/// @notice Contract to sell excess assets for USDC at a discount
+contract AaveV2CollectorContractConsolidation {
+    using SafeERC20 for ERC20;
+
+    uint256 public immutable USDC_DECIMALS;
+    uint256 public immutable ETH_USD_ORACLE_DECIMALS;
+
+    ERC20 public constant USDC = ERC20(TokenAddresses.USDC);
+    address public constant ETH_USD_FEED = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
+
+    event Purchase(address indexed tokenIn, address indexed tokenOut, uint256 amountIn, uint256 amountOut);
+
+    /// Not enough token left to purchase
+    error NotEnoughTokens(uint256 tokensLeft);
+    /// Oracle price is 0 or lower
+    error InvalidOracleAnswer();
+    /// Need to request more than 0 tokens out
+    error OnlyNonZeroAmount();
+    /// Token is not available for purchase
+    error UnsupportedToken(address token);
+
+    struct Asset {
+        uint256 quantity;
+        uint48 premium;
+        address oracle;
+        uint8 decimals;
+        uint8 oracleDecimals;
+        bool ethFeedOnly;
+    }
+
+    mapping(address => Asset) public assets;
+
+    constructor() {
+        assets[TokenAddresses.ARAI] = Asset(
+            ERC20(TokenAddresses.ARAI).balanceOf(AaveV2Ethereum.COLLECTOR),
+            100,
+            0x483d36F6a1d063d580c7a24F9A42B346f3a69fbb,
```

Comment:
The price feed addresses should be set as constants (in lib much better). This way will be much better to understand what they are, and much easier to read / check